File: FindDigits.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */   
    int t;
    long long int n;
    cin>>t;
    while(t--)
        {
        cin>>n;
        long long int num = n;
        int count = 0;
        while(n != 0)
            {
            int k = n % 10;
            n = n / 10;
            if(k == 0)
                continue;
            if(num % k == 0)
                count++;
            
        }
        cout<<count<<endl;
    }
    return 0;
}

File: FindDigits.go
package main
import "fmt"

func main() {
 //Enter your code here. Read input from STDIN. Print output to STDOUT
    var t int
    var n int
    var sum int
    fmt.Scanf("%v",&t)
    for i:=0;i<t;i++{
        sum = 0
        fmt.Scanf("%v",&n)
        var number = n
        for ;n!=0;{
            var remainder int
            remainder = n % 10
            n /= 10
            if remainder != 0 && number % remainder == 0{
                sum += 1
            }
        }
        fmt.Println(sum)
    }
}
